torch/csrc/jit/codegen/cuda/ir_interface_nodes.h

```diff
@@ -404,14 +404,20 @@ class TORCH_CUDA_CU_API TensorView : public Val {
       TensorView* var,
       TensorView* n);
 
-  // Create a TensorView before the original tensor. A common use case is to
-  // write results into shared memory or registers before moving to global
-  // memory. Analogous to TVM Cache_Write
-  TensorView* cache_before();
-
-  // Create a TensorView after the original tensor. A common use case is to
-  // read tensor into shared memory or registers. Analogous to TVM Cache_Read
-  TensorView* cache_after();
+  //! Create a TensorView before the original tensor. A common use case is to
+  //! write results into shared memory or registers before moving to global
+  //! memory. Analogous to TVM Cache_Write
+  //!
+  //! @param cache_op: memory operator to use for the inserted op between
+  //!   the the data tensor and the cache tensor
+  TensorView* cache_before(UnaryOpType cache_op = UnaryOpType::Set);
```

Comment:
`cache_op` seems to make these functions more overloaded than originally anticipated. We would probably need to do something again for Hopper TMA. I'd say this would be fine for now but we may want to revisit again in the future.

Reply:
The current cases seem to be at boundary to me. My understanding is that `cache_after` and `cache_before` were intended to model memory reads and writes, see the comment that were originally added above these lines. 

That's very much true in this use case except that we are just adding a bit more detail on "how" we are doing the read and write (i.e. using hardware intrinsics). And everything else about `cache_after` and `cache_before` still apply here. 

But I do agree that we could consider adding a separate interface for hardware accelerated memory IO if we keep adding to this. 

Let's postpone Hopper discussion until the official SDK release.